geode-core/src/distributedTest/java/org/apache/geode/management/JMXMBeanReconnectDUnitTest.java

```diff
@@ -36,117 +54,284 @@
+  public SerializableTemporaryFolder temporaryFolder = new SerializableTemporaryFolder();
 
-  private MBeanServerConnectionRule jmxConToLocator1;
-  private MBeanServerConnectionRule jmxConToLocator2;
+  @Rule
+  public SharedErrorCollector errorCollector = new SharedErrorCollector();
 
   @Rule
-  public ConcurrencyRule concurrencyRule = new ConcurrencyRule();
+  public transient GfshCommandRule gfsh = new GfshCommandRule();
 
   @Before
-  public void before() throws Exception {
-    locator1 = lsRule.startLocatorVM(0);
-    locator2 = lsRule.startLocatorVM(1, locator1.getPort());
-
-    server1 = lsRule.startServerVM(2, locator1.getPort());
-    // start an extra server to have more MBeans, but we don't need to use it in these tests
-    lsRule.startServerVM(3, locator1.getPort());
+  public void setUp() throws Exception {
+    locator1VM = getVM(1);
+    locator2VM = getVM(2);
+    serverVM = getVM(0);
+
+    locator1Name = "locator1";
+    locator2Name = "locator2";
+    serverName = "server1";
+    File locator1Dir = temporaryFolder.newFolder(locator1Name);
+    File locator2Dir = temporaryFolder.newFolder(locator2Name);
+    File serverDir = temporaryFolder.newFolder(serverName);
+
+    int[] port = getRandomAvailableTCPPorts(4);
+    locator1Port = port[0];
+    locator2Port = port[1];
+    locator1JmxPort = port[2];
+    locator2JmxPort = port[3];
+    locators = "localhost[" + locator1Port + "];localhost[" + locator2Port;
+
+    locator1VM.invoke(
+        () -> startLocator(locator1Name, locator1Dir, locator1Port, locator1JmxPort, locators));
+    locator2VM.invoke(
+        () -> startLocator(locator2Name, locator2Dir, locator2Port, locator2JmxPort, locators));
+
+    serverVM.invoke(() -> startServer(serverName, serverDir, locators));
+
+    gfsh.connectAndVerify(locator1JmxPort, PortType.jmxManager);
+
+    String regionName = "region1";
+    String createRegionCommand = "create region --type=REPLICATE --name=" + SEPARATOR + regionName;
+    gfsh.executeAndAssertThat(createRegionCommand).statusIsSuccess();
+
+    addIgnoredException(CacheClosedException.class);
+    addIgnoredException(CancelException.class);
+    addIgnoredException(DistributedSystemDisconnectedException.class);
+    addIgnoredException(ForcedDisconnectException.class);
+    addIgnoredException(MemberDisconnectedException.class);
+    addIgnoredException("Possible loss of quorum");
+
+    serverMXBeans = serverVM.invoke(() -> {
+      await().untilAsserted(() -> {
+        assertThat(getPlatformMBeanServer().queryNames(getInstance("GemFire:*"), null))
+            .containsAll(expectedMXBeansOnServer());
+      });
+      return getPlatformMBeanServer().queryNames(getInstance("GemFire:*"), null);
+    });
 
-    gfsh.connectAndVerify(locator1);
-    gfsh.executeAndAssertThat("create region --type=REPLICATE --name=" + REGION_PATH
-        + " --enable-statistics=true").statusIsSuccess();
+    mxbeansOnLocator1 = locator1VM.invoke(() -> {
```

Comment:
There are a lot of collapsible lambdas. I presume you intentionally did not collapse them for readability.